### PR TITLE
Math lib

### LINKED ISSUE #204: Should jou have a math lib?
python, c, c++ have math lib...
```python
# python
import math
```
```c
// C
#include <math.h>
```
```c++
// Cplusplus
#include <cmath>
```
So, does jou need a mathlib in future?
```python
# Jou
from ".../math.jou" import ...
```

### PR DESCRIPTION
**Pull request by littlewhitecloud**
write a math lib for jou...

fixes #204

### CODE REVIEW

**Comment by littlewhitecloud** [comment]
I haven't discard the last commit!

**Comment by Akuli** [comment]
Can you add a few tests? They could simply print stuff like `sin(12.34)`.

**Comment by littlewhitecloud** [comment]
okay, I will add

**Comment by littlewhitecloud** [comment]
llabs failed #216

**Comment by littlewhitecloud** [comment]
...how to run the check successfully?

**Comment by littlewhitecloud** [comment]
fixed llabs

**Comment by littlewhitecloud** [comment]
checks have passed, maybe it can merged?

File: tests/should_succeed/mathlibtest.jou [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    printf("%f\n", ceil(3.14159265358979)) # Output: 4.000000
+    printf("%f\n", floor(3.14159265358979)) # Output: 3.000000
+    printf("%f\n", round(3.14159265358979)) # Output: 3.000000
+    printf("%f\n", acos(-1)) # Output: 3.141593
+    printf("%f\n", asin(-1)) # Output: -1.570796
+    printf("%f\n", atan(-1)) # Output: -0.785398
+    printf("%f\n", atan2(-1, -1)) # Output: -2.356194
+    printf("%f\n", cos(45)) # Output: 0.525322
+    printf("%f\n", sin(45)) # Output: 0.850904
+    printf("%f\n", tan(90)) # Output: -1.995200
```
Comment (Akuli) [inline tests/should_succeed/mathlibtest.jou:23]: Please change the values to something else than 45, 45, 90. These functions take the angles as radians, so `cos(45)` is **not** the cosine of 45 degrees (which is sqrt(2)/2 so approximately 0.7, not 0.52).

File: stdlib/math.jou [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# -*- mathlib -*-
+# abs, llabs, fabs, ceil, floor, round, acos, asin, atan, atan2, cos, sin, tan, fmod, sqrt, cbrt, log, log10, log2, exp, exp2, pow, ldexp, fmax, fmin, hypot
+
```
Comment (Akuli) [inline stdlib/math.jou:3]: We don't need these comments:

```suggestion
```

File: tests/should_succeed/mathlibtest.jou [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# This is a math lib test for math lib
+# It includes abs, llabs, fabs, ceil, floor, round, acos, asin, atan, atan2, cos, sin, tan,
+#             fmod, sqrt, cbrt, log, log10, log2, exp, exp2, pow, ldexp, fmax, fmin, hypot
```
Comment (Akuli) [inline tests/should_succeed/mathlibtest.jou:3]:
```suggestion
```

**Review by Akuli — COMMENTED** [review]
Thanks!

File: tests/should_succeed/mathlibtest.jou (hunk 1 above)
Comment (littlewhitecloud) [reply tests/should_succeed/mathlibtest.jou:23]: changed

**Comment by littlewhitecloud** [comment]
okay, I changed the test arg, and all the check are successful…

**Comment by littlewhitecloud** [comment]
I don’t know why the pull request doesn’t up to date to my main branch

**Comment by littlewhitecloud** [comment]
So I created a new one

**Pull request closed without merging**